src/platform/bouffalolab/BL602/wifi_mgmr_portable.c

```diff
@@ -103,3 +112,39 @@ int wifi_mgmr_profile_ssid_get(uint8_t * ssid)
 
     return profile_msg.ssid_len;
 }
+
+uint32_t wifi_mgmr_get_security_type(void)
+{
+    /** return values defined from app::Clusters::WiFiNetworkDiagnostics::WiFiVersionEnum */
+    if (strlen(wifiMgmr.wifi_mgmr_stat_info.passphr))
+    {
+        if (gWpaSm.pmk_len)
+        {
+            if (WPA_PROTO_WPA == gWpaSm.proto)
+            {
+                return 3;
```

Comment:
`WiFiVersionEnum::kXX `value be used here instead of the number?

```
  enum WiFiVersionEnum : ENUM8 {
    kA = 0;
    kB = 1;
    kG = 2;
    kN = 3;
    kAc = 4;
    kAx = 5;
  }
```

Reply:
enum is defined for CPP with namespace protected. Here is a C file, so I use the number to return.

Reply:
I do think that a comment in this function indicating the return values need to match that enum would be useful.

Reply:
Comment was corrected from WiFiVersionEnum to SecurityTypeEnum.

Reply:
The function returns the security type.
Comment was corrected to indicate the return value from enum SecurityTypeEnum.

Reply:
The return values are still "magic numbers" and it is not easily readeable what is being return. Also, if the spec change values for whatever reason that would break. 

Any reason what you implementing c files for matter platform code? Matter is implemented in c++, being able to use the different namespace and utilities for matter in your matter platform code would be beneficial

Reply:
Our SDK is based on C lanuage, and including these C headers in CPP files will make compile error.
I changed C file in our Matter platform code to wraper security type accessing; and determined SecurityTypeEnum in DiagnosticDataProviderImpl.cpp with these wraper APIs.